dev/devicelab/bin/tasks/flutter_engine_group_performance.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/task_result.dart';
+
+const String _pathSeperator = '/';
+const String _bundleName = 'dev.flutter.multipleflutters';
+const String _activityName = 'MainActivity';
+
+List<String> _splitAbsPath(String path) {
+  if (path[0] != '/') {
+    throw Exception('expected absolute path');
+  }
+  assert(path[0] == '/');
+  final List<String> result = path.split(_pathSeperator);
+  result.removeWhere((String element) => element.isEmpty);
+  return result;
+}
+
+String _joinAbsPath(List<String> components) {
+  return _pathSeperator + components.join(_pathSeperator);
+}
+
+List<String> _addComponent(List<String> components, String value) {
+  final List<String> result = <String>[];
+  result.addAll(components);
+  result.add(value);
+  return result;
+}
+
+List<String> _addComponents(List<String> components, List<String> values) {
+  final List<String> result = <String>[];
+  result.addAll(components);
+  result.addAll(values);
+  return result;
+}
+
+Future<void> _run(String command, List<String> args, String cwd) async {
+  print('$command: $args');
+  final Process process =
+      await Process.start(command, args, workingDirectory: cwd);
+  final Future<dynamic> stdoutStream = stdout.addStream(process.stdout);
+  final Future<dynamic> stderrStream = stderr.addStream(process.stderr);
+  final int exitCode = await process.exitCode;
+  if (exitCode != 0) {
+    throw Exception('command "$command $args" had exit code: $exitCode');
+  }
+  await stdoutStream;
+  await stderrStream;
+}
+
+Future<TaskResult> _doTest() async {
+  try {
+    final List<String> scriptPathComponents =
+        _splitAbsPath(Platform.script.path);
+    final List<String> multipleFluttersPath =
+        scriptPathComponents.sublist(0, scriptPathComponents.length - 4);
+    multipleFluttersPath.addAll(<String>['benchmarks', 'multiple_flutters']);
+    final String modulePath =
+        _joinAbsPath(_addComponent(multipleFluttersPath, 'module'));
+    final String androidPath =
+        _joinAbsPath(_addComponent(multipleFluttersPath, 'android'));
```

Comment:
Use `package:path/path.dart` instead of implementing path helpers yourself.  There's also utilities methods used by other devicelab tests.

```dart
import 'package:path/path.dart' as path;
import 'package:flutter_devicelab/framework/adb.dart';
import 'package:flutter_devicelab/framework/utils.dart';

...
final String multipleFluttersPath = path.join(flutterDirectory.path, 'dev', 'benchmarks', 'multiple_flutters');
final String modulePath = path.join(multipleFluttersPath, 'module');

await inDirectory(modulePath, () async {
    await flutter('pub', options: <String>['get']);
});
```
etc

Reply:
done